src/manager.rs

```diff
@@ -0,0 +1,194 @@
+use compile::ExecutableDevEnv;
+use parse::Parser;
+use run::{ Execution, ExecutionEvent, Error as RunError };
+use rusqlite;
+use serde_json::error::Error as SerdeJsonError;
+use std::collections::HashMap;
+use std::io::Error as IOError;
+use std::path::{ Path, PathBuf };
+use std::sync::mpsc::RecvError;
+use transformable_channels::mpsc::{ channel, ExtSender };
+
+
+/// A ScriptManager error.
+#[derive(Debug)]
+pub enum Error {
+    NoSuchScriptError,
+    ScriptAlreadyRunning,
+    RusqliteError(rusqlite::Error),
+    IoError(IOError),
+    ExecutionError(ExecutionEvent),
+    RunError(RunError),
+    JsonError(SerdeJsonError),
+    RecvError(RecvError)
+}
+
+/// ScriptManager stores a persistent database of scripts and executes them.
+/// Each script can be individually enabled or disabled.
+/// When a script is enabled, it is always running (unless an error occured during launch).
+pub struct ScriptManager<Env> where Env: ExecutableDevEnv + 'static {
+    path: PathBuf,
+    env: Env,
+    runners: HashMap<String, Execution<Env>>
```

Comment:
Let's replace these `String` with a `Id<ScriptId>`. You'll need to define an empty `pub enum` or `pub struct` `ScriptId`, just for the sake of type-safety. This will save us lots of memory and copies a bit later.